pkg/designate/generate_bind9_pools_yaml.go

```diff
@@ -0,0 +1,215 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package designate
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+// PoolData represents the data GeneratePoolsYamlFile will receive from the controller
+type PoolData struct {
+	Attributes   map[string]string
+	BindMap      map[string]string
+	MdnsMap      map[string]string
+	NsRecordsMap map[string]int
+}
+
+type Pool struct {
+	Name        string
+	Description string
+	Attributes  map[string]string
+	NSRecords   []NSRecord
+	Nameservers []Nameserver
+	Targets     []Target
+	CatalogZone *CatalogZone // it is a pointer because it is optional
+}
+
+type NSRecord struct {
+	Hostname string
+	Priority int
+}
+
+type Nameserver struct {
+	Host string
+	Port int
+}
+
+type Target struct {
+	Type        string
+	Description string
+	Masters     []Master
+	Options     Options
+}
+
+type Master struct {
+	Host string
+	Port int
+}
+
+type Options struct {
+	Host           string
+	Port           int
+	RNDCHost       string
+	RNDCPort       int
+	RNDCConfigFile string
+}
+
+type CatalogZone struct {
+	FQDN    string
+	Refresh string
+}
+
+func GeneratePoolsYamlFile(pooslData []PoolData) (string, error) {
```

Comment:
probably meant poolsData

Reply:
Yes, I don't return a file, thanks.
Maybe poolsYamlContent? I return a string that could be saved as the pools.yaml file.

Reply:
I will change it to poolsYamlData